dask/dataframe/core.py

```diff
@@ -75,12 +76,25 @@ def _get_return_type(meta):
     return Scalar
 
 
-def new_dd_object(dsk, _name, meta, divisions):
+def new_dd_object(dsk, name, meta, divisions):
     """Generic constructor for dask.dataframe objects.
 
     Decides the appropriate output class based on the type of `meta` provided.
     """
-    return _get_return_type(meta)(dsk, _name, meta, divisions)
+    if isinstance(meta, pd.core.base.PandasObject):
```

Comment:
Is this class public enough to use here? In other places we've done `(pd.Series, pd.Index, pd.DataFrame)`.